shared-bindings/bitmaptools/__init__.c

```diff
@@ -154,8 +161,8 @@ STATIC mp_obj_t bitmaptools_obj_rotozoom(size_t n_args, const mp_obj_t *pos_args
           ARG_angle, ARG_scale, ARG_skip_index};
 
     static const mp_arg_t allowed_args[] = {
-        {MP_QSTR_dest_bitmap, MP_ARG_REQUIRED | MP_ARG_OBJ},
-        {MP_QSTR_source_bitmap, MP_ARG_REQUIRED | MP_ARG_OBJ},
+        {MP_QSTR_dest_bitmap, MP_ARG_REQUIRED | MP_ARG_OBJ, {.u_obj = 0}},
+        {MP_QSTR_source_bitmap, MP_ARG_REQUIRED | MP_ARG_OBJ, {.u_obj = 0}},
```

Comment:
The default values should not be necessary if the arg is required. Is the compiler complaining/

Reply:
Yes.  There's a diagnostic flag that seems to be enabled on the unix build but not the others.  Typical diagnostic text:
```
      |                                                                        ^~~~~~~~~~~~~~
../../shared-bindings/bitmaptools/__init__.c: In function ‘bitmaptools_dither’:
../../shared-bindings/bitmaptools/__init__.c:685:9: error: missing initializer for field ‘defval’ of ‘mp_arg_t’ {aka ‘const struct _mp_arg_t’} [-Werror=missing-field-initializers]
  685 |         { MP_QSTR_dest_bitmap, MP_ARG_REQUIRED | MP_ARG_OBJ },
      |         ^
In file included from ../../shared-bindings/bitmaptools/__init__.c:37:
../../py/runtime.h:61:18: note: ‘defval’ declared here
   61 |     mp_arg_val_t defval;
      |                  ^~~~~~
```

Reply:
I looked at micropython, and they do indeed add a value for this field, except in a few routines in some ports (which are probably old). However, they use `MP_OBJ_NULL` instead of a `0`. Maybe use that just for consistency.

I wonder if we should turn this on in general. It might break a bunch of things, though. Maybe that's what you already discovered :)